lib/matplotlib/axes/_base.py

```diff
@@ -2949,6 +2949,8 @@ def tick_params(self, axis='both', **kwargs):
         also be red.  Gridlines will be red and translucent.
 
         """
+        if axis not in ['x', 'y', 'both']:
+            raise ValueError("axis must be one of 'x', 'y' or 'both'.")
```

Comment:
no dot at end of exception messages (that's both what we do and consistent with the stdlib)